### PR TITLE
MAC Address Mapping when adding new Pis to Piman

### PR DESCRIPTION
**Pull request by markbrogan13**
This code tweak will allow the logging the the DHCP runtime:

1. MAC addresses that are new, will now display the company that they came from
1. Added a CSV database where the MAC grabbed will be referenced to

Follows closely to how wireshark knows what company the device comes from, without needing wireshark installed.

### CODE REVIEW

**Review by ajones239 — APPROVED** [review]
I'm Aidan Jones, and I approve this pull request.

**Review by aldogutierrez — APPROVED** [review]
Reviewed & approved MAC address mapper 👍🏽 

- Aldo Gutiérrez

File: dhcp/dhcp.py [hunk 1]
```diff
@@ -192,7 +193,16 @@ def received_dhcp_discover(self, discovery):
         if should_send_offer:
             self.send_offer(discovery)
         else:
-            logger.error("unknown mac_address {}. will not assign ip.".format(discovery.client_mac_address))
+            unknown_mac_addr = discovery.client_mac_address
+            logger.error("unknown mac_address {}. will not assign ip.".format(unknown_mac_addr))
+            self.mac_mapper(unknown_mac_addr)
+
+    def mac_mapper(self, unknown_mac):
+        with open('addr_database.csv', 'r') as f:
```
Comment (breed) [inline dhcp/dhcp.py]: you should pull this from the pyz like tftp does. also you should check in this file and document how to update it.

File: dhcp/dhcp.py [hunk 2]
```diff
@@ -192,7 +193,16 @@ def received_dhcp_discover(self, discovery):
...
+            reader - csv.reader(f, delimiter=',')
+            for row in reader:
+                if row[0] in unknown_mac:
```
Comment (breed) [inline dhcp/dhcp.py]: i think you want to check startswith().

**Review by breed — CHANGES_REQUESTED** [review]
two code changes and the addition of a file

File: dhcp/dhcp.py (hunk 1 above)
Comment (markbrogan13) [reply dhcp/dhcp.py]: Hi Ben, could you elaborate on what you mean by pulling from the pyz like tftp does? My adaptation just replaces what happens when you start looking for DHCP discovers

Comment (breed) [reply dhcp/dhcp.py]: addr_database.csv is a resource, so it should be built into the pyz file and accessed like a resource: https://github.com/SJSU-CS-systems-group/piman/blob/master/tftp/tftp.py#L43

Comment (markbrogan13) [reply dhcp/dhcp.py]: added new commit to resolve this issue

File: dhcp/dhcp.py (hunk 2 above)
Comment (markbrogan13) [reply dhcp/dhcp.py]: changed with most recent commit

File: dhcp/dhcp.py (hunk 1 above)
Comment (breed) [reply dhcp/dhcp.py]: sorry, where is the commit?

**Review by breed — CHANGES_REQUESTED** [review]

File: dhcp/dhcp.py (hunk 1 above)
Comment (markbrogan13) [reply dhcp/dhcp.py]: https://github.com/SJSU-CS-systems-group/piman/pull/13/commits/4724ba7744dc13781ee28dea2da8bc293820581a

**Comment by markbrogan13** [comment]
https://github.com/SJSU-CS-systems-group/piman/pull/13/commits/4724ba7744dc13781ee28dea2da8bc293820581a

This has the two most recent commits that fix both of the changes requested

File: dhcp/dhcp.py [hunk 3]
```diff
@@ -198,7 +198,7 @@ def received_dhcp_discover(self, discovery):
             self.mac_mapper(unknown_mac_addr)
 
     def mac_mapper(self, unknown_mac):
-        with open('addr_database.csv', 'r') as f:
+        with open('install/addr_database.csv', 'r') as f:
```
Comment (breed) [inline dhcp/dhcp.py]: this needs to be pulled from the pyz file.

**Review by breed — CHANGES_REQUESTED** [review]

**Comment by markbrogan13** [comment]
https://github.com/SJSU-CS-systems-group/piman/pull/13/commits/b4a54a58b024a2204b477f800a9af03b31da3a41

With help from Aidan, imitated the way tftp.py does the file read in the pyz file

**Review by breed — APPROVED** [review]
awesome! thanks!

**Pull request merged**